shared/chat/inbox/header/container.tsx

```diff
@@ -42,9 +41,9 @@ export default namedConnect(
     onSelectDown: dispatchProps.onSelectDown,
     onSelectUp: dispatchProps.onSelectUp,
     showFilter: stateProps.showFilter,
-    showNewChat: ownProps.showNewChat,
-    showSearch: ownProps.showSearch,
-    showStartNewChat: stateProps.showNewChat,
+    showNewChat: ownProps.context == 'chat-header',
+    showSearch: ownProps.context == 'chat-header' ? !isTablet : isMobile,
+    showStartNewChat: stateProps.showStartNewChat,
```

Comment:
These should have the same values as before. It's easier for me to read when these boolean decisions are all in the same file.